src/libstd/fs.rs

```diff
@@ -674,6 +716,41 @@ impl DirEntry {
     /// The exact text, of course, depends on what files you have in `.`.
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn path(&self) -> PathBuf { self.0.path() }
+
+    /// Return the metadata for the file that this entry points at.
+    ///
+    /// This function may not require an extra system call on some platforms
```

Comment:
Good idea! I've started out with the platform behavior sections to see where it takes us.